### DISCUSSION TITLE
Optimizing Optical Design Problems with Constraints

### DISCUSSION THREAD

**Discussion by alexbrownsoft** [ceres-solver/ceres-solver#959] [Q&A] [answered]
I am trying to find a tool to carry out optimization for optical len design problems, such as to minimize a non-linear merit function with other non-linear constraints. The variables are also with in bounded  range.

Can I use ceres solver for this purpose?

**Comment by sandwichmaker** [comment, marked as answer]
@alexbrownsoft I suspect what you are looking for is closer to IPOPT rather than ceres solver.

https://coin-or.github.io/Ipopt/index.html#Overview
